drivers/ethernet/eth_sam_gmac_priv.h

```diff
@@ -142,11 +178,30 @@
 		(GMAC_IERPQ_RCOMP | GMAC_INTPQ_RX_ERR_BITS | \
 		 GMAC_IERPQ_TCOMP | GMAC_INTPQ_TX_ERR_BITS | GMAC_IERPQ_HRESP)
 
+#if GMAC_QUEUE_NUM > 1
+	/* 4 kB Receiver Packet Buffer Memory Size */
+	/* 4 kB Transmitter Packet Buffer Memory Size */
+	/* Transmitter Checksum Generation Offload Enable */
+#define GMAC_DMA_QUEUE_FLAGS \
+		(GMAC_DCFGR_RXBMS_FULL | GMAC_DCFGR_TXPBMS | \
+		 GMAC_DCFGR_TXCOEN)
+#else
+#define GMAC_DMA_QUEUE_FLAGS (0)
+#endif
+
 /** List of GMAC queues */
 enum queue_idx {
 	GMAC_QUE_0,  /** Main queue */
+#if !defined(CONFIG_SOC_SERIES_SAM4E)
 	GMAC_QUE_1,  /** Priority queue 1 */
 	GMAC_QUE_2,  /** Priority queue 2 */
+#endif
+#if defined(CONFIG_SOC_ATMEL_SAME70_REVB) || \
+	defined(CONFIG_SOC_ATMEL_SAMV71_REVB)
+	GMAC_QUE_3,  /** Priority queue 3 */
+	GMAC_QUE_4,  /** Priority queue 4 */
+	GMAC_QUE_5,  /** Priority queue 5 */
+#endif
 };
```

Comment:
This enum might have outlived its usefulness. The only place where it's used now is the initialization of `queue_list`, as in
```
	.que_idx = GMAC_QUE_3,
```
Here, an integer literal would be just as readable.

Reply:
I prefer keep queue_idx for now but removed all preprocessor definitions. There are places that uses GMAC_QUE_1 and GMAC_QUE_2 and maybe we need review it. I took liberty to add some GMAC_QUE_x uses. Anyway, if will be need drop it I will not complain.